autoload/suda.vim

```diff
@@ -15,6 +15,7 @@ function! suda#init(...) abort
 endfunction
 
 function! suda#system(cmd, ...) abort
+  let sudo_needs_a_password = system('sudo -v -n')
```

Comment:
It seems you are trying to validate with a cache. However, if a cache is enabled, `system('sudo -p '''' -n {cmd}')` at line 23 should success. So it seems the cache is not available when called with `system()` function.

Reply:
I'll explain in more detail what I've explained in the 1st comment in this PR:
> Users who have timestamp_type set to global in /etc/sudoers can
use the sudo cache inorder not to repeat the password over and over
again when saving.
Try putting the following in your `/etc/sudoers`:
```
Defaults timestamp_type = global
```
Please do it and then test my PR. It won't have any effect without this.

Reply:
Unfortunately

<img width="364" alt="1 visudo 2018-07-14 01-51-04" src="https://user-images.githubusercontent.com/546312/42703730-6012823c-8708-11e8-973a-c14301105d1f.png">

Reply:
What version of `sudo` do you have installed?

Reply:
```
Sudo version 1.8.17p1
Sudoers policy plugin version 1.8.17p1
Sudoers file grammar version 45
Sudoers I/O plugin version 1.8.17p1
```

on macOS High Sierra

Reply:
Hmm, I'm trying to understand from the [change-log of sudo](https://www.sudo.ws/devel.html) if this option was introduced in a recent version or something alike but I've found no clue. It could be it's not available on macOS from some reason... Could you check for me if you can see it in the manual `sudoers(5)` installed on your machine?

BTW the current version of `sudo` is `1.8.22`.

Reply:
> Could you check for me if you can see it in the manual sudoers(5) installed on your machine?

Yes but no `timestamp_type`